test/make/signature.t

```diff
@@ -0,0 +1,122 @@
+---------------------------------------------------
+NOTICE: @@ocaml.doc and @@merlin.hide annotations
+& `include struct` boilerplate are added by ppxlib.
+---------------------------------------------------
+
+Test 1: Given a regular record type a, expose make_a
+  $ test1="
+  > type a = {
+  >   x: int ;
+  >   y: bool }[@@deriving make]"
+  $ echo "$test1" > test.mli
+  $ driver test.mli 
+  type a = {
+    x: int ;
+    y: bool }[@@deriving make]
+  include sig [@@@ocaml.warning "-32"] val make_a : x:int -> y:bool -> a end
+  [@@ocaml.doc "@inline"][@@merlin.hide ]
+
+Test 2: Given a nonrec type, embed error
+  $ test2="
+  > type nonrec b = {
+  >   x: int ;
+  >   y: bool }[@@deriving make]"
+  $ echo "$test2" > test.mli
+  $ driver test.mli
+  type nonrec b = {
+    x: int ;
+    y: bool }[@@deriving make]
+  include
+    sig
+      [@@@ocaml.warning "-32"]
+      [%%ocaml.error "nonrec is not compatible with the `make' preprocessor."]
+    end[@@ocaml.doc "@inline"][@@merlin.hide ]
+
+Test 3: Given a non-record type, embed error
+  $ test3="
+  > type c = int * int
+  > [@@deriving make]"
+  $ echo "$test3" > test.mli
+  $ driver test.mli
+  type c = (int * int)[@@deriving make]
+  include
+    sig
+      [@@@ocaml.warning "-32"]
+      [%%ocaml.error
+        "Unsupported use of make (you can only use it on records)."]
+    end[@@ocaml.doc "@inline"][@@merlin.hide ]
+
+Test 4: Given a private type, embed error
```

Comment:
I think we talked about this before and I just said "just fail on private types", but now I'm wondering: why would we fail on private types? Declaring a type private and using the make deriver on it seems like a good option to have a pretty much "abstract" type and deriving a make function for it. Let me know if what I'm saying is unclear!

Reply:
What's failing here is the signature deriver -- the deriver will work with the structure of private types (see test 4 in `structure.t`. So the error is about exposing the private types. But are you saying it still shouldnt fail for the signature? @pitag-ha

Reply:
Tbh, I'm not very familiar with private types myself. As far as I remember, it's possible and ok to expose private types. If that's the case, I think it would be nice to be able to apply the deriver to that. If you want, we can look into what exactly private types are together one of these days!

Reply:
Ok! I just removed the private types check and promoted the tests. There is no longer a special case for private types

Reply:
Cool, thanks!